Problem Statement :
The manufacturer of Print-Heads is concerned about how good their products work and wants to find the Performance(How many prints(in millions) each machine can do).

In [1]:
import pandas as pd
import numpy as np
from scipy import stats

We have the sample data of 15 machines:
1.13, 1.55, 1.43, 0.92, 1.25, 1.36, 1.32, 0.85, 1.07, 1.48, 1.20, 1.33, 1.18, 1.22, 1.29

In [2]:
Sample_data=np.array([1.13, 1.55, 1.43, 0.92, 1.25, 1.36, 1.32, 0.85, 1.07, 1.48, 1.20, 1.33, 1.18, 1.22, 1.29])
Sample_data

array([1.13, 1.55, 1.43, 0.92, 1.25, 1.36, 1.32, 0.85, 1.07, 1.48, 1.2 ,
       1.33, 1.18, 1.22, 1.29])

In [7]:
n=len(Sample_data)

In [5]:
sample_mean=np.mean(Sample_data)
sample_std=np.std(Sample_data)
population_std=0.2 #Given in Case-2
sample_mean,sample_std

(1.2386666666666666, 0.18661427836285438)

Case-1

In [ ]:
'''In Case-1 
We got have sample_data and sample_mean, sample_standard_deviation is calculated but we dont know the population standard deviation
And also our Sample_data size is small we go with the T-distribution '''

In [ ]:
'''Formula for confidence intervals with T-distribution is [point_estimator-margin_of_error,point_estimator+margin_of_error]
Here Margin_of_erro= Critical_value(T)*standard_deviation(sample_data) / square_root(No.of Values)'''

In [8]:
'Case-1'
confidence_level=0.99
Alpha_value=1-confidence_level
DOF=n-1
T_critical=stats.t.ppf(1-Alpha_value/2,df=DOF)
Margin_of_error=T_critical*(sample_std/np.sqrt(n))
confidence_intervals=[sample_mean-Margin_of_error,sample_mean+Margin_of_error]

In [9]:
confidence_intervals

[1.095231668626123, 1.3821016647072102]

In [13]:
np.mean(confidence_intervals)

1.2386666666666666

We are 99% confident that the True mean(actual mean) of the print heads will lies between these intervals [1.095231668626123, 1.3821016647072102] (Values in millions)

.

In [ ]:
'''In Case-2 
Here we got all the values we required including Population standard deviation which is 0.2
So we will go with the Z-distribution'''

In [ ]:
'''Formula for confidence intervals with Z-distribution is [point_estimator-margin_of_error,point_estimator+margin_of_error]
Here Margin_of_erro= Critical_value(Z)*standard_deviation(sample_data) / square_root(No.of Values)'''

In [14]:
'Case-2'
confidence_level_=0.99
z_critical=stats.norm.ppf(1-Alpha_value/2)
Margin_of_error_Z=z_critical*(population_std/np.sqrt(n))
confidence_intervals_z=[sample_mean-Margin_of_error_Z,sample_mean+Margin_of_error_Z]

In [15]:
confidence_intervals_z

[1.1056514133957607, 1.3716819199375725]

We are 99% confident that the True mean(actual mean) of the print heads will lies between these intervals [1.1056514133957607, 1.3716819199375725] (Values in millions) .